src/formatDate.js

```diff
@@ -7,8 +7,47 @@
  *
  * @returns {string}
  */
+
+const DIVIDER_IDX = 3;
+const YEARS_PERIOD = 30;
+const YEARS_2000S = 20;
+const YEARS_1900S = 19;
+const YEAR_SHORT = 'YY';
+const YEAR_LONG = 'YYYY';
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const divider = fromFormat[DIVIDER_IDX];
+  const dateParts = date.split(divider);
+  const dateConfig = {};
+
+  for (let i = 0; i < 3; i++) {
+    const datePart = dateParts[i];
+    const formatPart = fromFormat[i];
+
+    if (formatPart === YEAR_LONG) {
+      dateConfig.YY = datePart.slice(2);
```

Comment:
When assigning the last two digits of the year to 'YY', you should also check if the 'YYYY' key in dateConfig is not already set to avoid overwriting it.